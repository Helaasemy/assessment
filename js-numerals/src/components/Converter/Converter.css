.input {
    margin: auto;
    position: absolute;
    height: 50px;
    width: 300px;
    top: 45%;
    left: 40vw;
}
.result {
    font-weight: bold;
    margin-top: 18px;
}
input {
    box-sizing: border-box;
    width: 100%;
    height: 50px;
    padding-left: 10px;
    border: 4px solid #ffffff;
    background: none;
    color: #fff;
    font-size: 16px;
    font-weight: 400;
    outline: 0;
}
input::placeholder {
    color: white;
}
