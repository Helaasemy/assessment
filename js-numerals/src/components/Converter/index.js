import React, { useState } from 'react';
import { LanguageConvert, InputNumConverter } from '../../utils';
import './Converter.css';

function Converter() {
    const [value, setValue] = useState('');
    const handleChange = ({ target }) => {
        setValue(target.value)
    }
    const number = LanguageConvert(value)
    const result = InputNumConverter(number)

    return (
        <div className="input">
            <input
                placeholder="write an arabic number"
                type="text"
                onChange={handleChange}
                data-testid="input"
            />
            <div className='result' data-testid="result" >
                {result}
            </div>
        </div>
    );
}

export default Converter