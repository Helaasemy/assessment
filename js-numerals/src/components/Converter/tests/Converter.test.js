import { render, fireEvent } from '@testing-library/react';
import Converter from '../index';

describe('<Converter/>', () => {

  it('should renders without errors', () => {

    const { container } = render(
      <Converter />
    );
    expect(container).toMatchSnapshot();
  });

  it('should return any number as words', () => {

    const { getByTestId } = render(
      <Converter />
    );
    fireEvent.change(getByTestId('input'), { target: { value: '٢٤' } })
    expect(getByTestId('result').textContent).toBe('twenty four ');

  });
  it('return not a number', () => {

    const { getByTestId } = render(
      <Converter />
    );
    fireEvent.change(getByTestId('input'), { target: { value: 'writing a text' } })
    expect(getByTestId('result').textContent).toBe('not a number');

  });
  it('return number too big if more than 15 digits', () => {

    const { getByTestId } = render(
      <Converter />
    );
    fireEvent.change(getByTestId('input'), { target: { value: '1234567890123456' } })
    expect(getByTestId('result').textContent).toBe('number is too big');

  });
});