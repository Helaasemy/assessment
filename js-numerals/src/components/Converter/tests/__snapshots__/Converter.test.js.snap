// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Converter/> should renders without errors 1`] = `
<div>
  <div
    class="input"
  >
    <input
      data-testid="input"
      placeholder="write an arabic number"
      type="text"
    />
    <div
      class="result"
      data-testid="result"
    />
  </div>
</div>
`;
